gremlin-core/src/test/java/org/apache/tinkerpop/gremlin/process/traversal/step/branch/RepeatStepTest.java

```diff
@@ -39,7 +39,9 @@
                 __.repeat(out()).times(3),
                 __.repeat(out().as("x")).times(3),
                 __.out().emit().repeat(out()).times(3),
-                __.repeat(out()).until(hasLabel("x"))
+                __.repeat(out()).until(hasLabel("x")),
+                __.repeat("a", __.out()).times(3),
```

Comment:
Sorry, perhaps I should have been more clear on what kind of traversals to add here.

Can you please also add `__.repeat(out()).times(3)`? Without the changes in `hashCode()` this traversal would have had the same hashCode as `__.repeat("a", out()).times(3)` and thus this test would have failed.

I guess these two will still collide:

```
__.repeat("a", __.repeat("b", out()))
__.repeat("b", __.repeat("a", out()))
```

...but I can't think of a solution that doesn't involve any expensive method calls.

Reply:
I don't quite understand your first point - `__.repeat(out()).times(3)` is the first entry in the array and I believe this did fail before the `hashCode()` fix.

Good point about the collision. 
Maybe I could use the same method the emit first and until first use? ie:
```
@Override
    public int hashCode() {
        int result = super.hashCode() ^ (this.repeatTraversal.hashCode() << 1);
        result ^= Boolean.hashCode(this.untilFirst);
        result ^= Boolean.hashCode(this.emitFirst) << 1;
        if (this.loopName != null)
            result ^= this.loopName.hashCode();
        if (this.untilTraversal != null)
            result ^= this.untilTraversal.hashCode();
        if (this.emitTraversal != null)
            result ^= this.emitTraversal.hashCode();
        return result;
    }
```

Reply:
Ah, alright, I totally missed the first entry, all good then.

We would have to shift or rotate the `loopName`'s hashCode value by the depth of the repeat traversal. Unfortunately, we don't have this information and I don't see how your `hashCode` implementation above would solve the problem. Anyway, it's an unlikely corner case and I don't think we need to care about it too much, so no big deal if we don't have a solution for now.